package/MDAnalysis/transformations/nojump.py

```diff
@@ -118,6 +118,14 @@ def _transform(self, ts):
         except np.linalg.LinAlgError:
             msg = f"Periodic box dimensions are not invertible at step {ts.frame}"
             raise NoDataError(msg)
+
+        # Ensure NoJump is applied from the first frame onwards
+        if ts.frame > 0 and self.prev is None:
+            raise ValueError(
```

Comment:
From the codecov comment below you can see that the `raise ValueError` is never triggered by the tests. Thus, you have to improve your test.